pkg/asset/installconfig/gcp/validation_test.go

```diff
@@ -110,6 +111,48 @@ var (
 	validateXpnSA            = func(ic *types.InstallConfig) { ic.ControlPlane.Platform.GCP.ServiceAccount = validXpnSA }
 	invalidateXpnSA          = func(ic *types.InstallConfig) { ic.ControlPlane.Platform.GCP.ServiceAccount = invalidXpnSA }
 
+	validCPKMSKeyRing = func(ic *types.InstallConfig) {
+		ic.ControlPlane.Platform.GCP.OSDisk = gcp.OSDisk{
+			EncryptionKey: &gcp.EncryptionKeyReference{
+				KMSKey: &gcp.KMSKeyReference{
+					Name:    "validKeyName",
+					KeyRing: "validKeyRingName",
+				},
+			},
+		}
+	}
+	invalidateCPKMSKeyRing = func(ic *types.InstallConfig) {
+		ic.ControlPlane.Platform.GCP.OSDisk = gcp.OSDisk{
+			EncryptionKey: &gcp.EncryptionKeyReference{
+				KMSKey: &gcp.KMSKeyReference{
+					Name:    "invalidKeyName",
+					KeyRing: "invalidKeyRingName",
+				},
+			},
+		}
+	}
+
+	validComputeKMSKeyRing = func(ic *types.InstallConfig) {
+		ic.Compute[0].Platform.GCP.OSDisk = gcp.OSDisk{
+			EncryptionKey: &gcp.EncryptionKeyReference{
+				KMSKey: &gcp.KMSKeyReference{
+					Name:    "validKeyName",
+					KeyRing: "validKeyRingName",
+				},
+			},
+		}
+	}
+	invalidateComputeKMSKeyRing = func(ic *types.InstallConfig) {
+		ic.Compute[0].Platform.GCP.OSDisk = gcp.OSDisk{
+			EncryptionKey: &gcp.EncryptionKeyReference{
+				KMSKey: &gcp.KMSKeyReference{
+					Name:    "validKeyName",
+					KeyRing: "invalidKeyRingName",
+				},
+			},
+		}
+	}
+
```

Comment:
Would be useful to add a test for just DefaultMachinePlatform. I think when a key is provided in DefaultMachinePlatform and one in Compute, we will actually end up using the key provided with the Compute. But during validation, if we encounter an invalid key within the DefaultMachinePlatform and a valid one in Compute, we would still fail the validation.